src/core/tsi/ssl_transport_security.cc

```diff
@@ -1952,6 +1989,22 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
     SSL_CTX_set_session_cache_mode(ssl_context, SSL_SESS_CACHE_CLIENT);
   }
 
+  if (options->key_logger != nullptr) {
+    // Unref is manually called on factory destruction
+    impl->key_logger =
+        reinterpret_cast<tsi::TlsKeyLogger*>(options->key_logger)->Ref();
+    if (options->session_cache == nullptr) {
+      // Need to set factory at g_ssl_ctx_ex_factory_index
+      SSL_CTX_set_ex_data(ssl_context, g_ssl_ctx_ex_factory_index, impl);
+    }  // Otherwise it is already set
+
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+       // SSL_CTX_set_keylog_callback is set here to register callback
+    // when ssl/tls handshakes complete.
+    SSL_CTX_set_keylog_callback(ssl_context, ssl_client_keylogging_callback);
```

Comment:
No its supposed to be set only when key_logger is not a nullptr. Key logger functions need to be accessible to append to files. Otherwise there is no need to set a keylog callback